### DISCUSSION TITLE
Does sync feature cause a possibility for data corruption in Zotero

### DISCUSSION THREAD

**Discussion by ulahcherubim** [windingwind/zotero-better-notes#859] [General]
I am very thankful to all contributors and especiallly @windingwind for this plugin. My workflow with Obsidian is quite simple now, and I can edit my notes on both programs. But I have a concern about two way sync between two programs. I am using a collection in Zotero to store my zettels/atomic notes and sync them with obsidian for editing. I want to make use of "related" field of Zotero to make the related items' title appear at the top of the note in brackets[[ ]]. This way, I can make Wikilinks via Zotero but when it syncs back it won't assume these links as related items. So actually if I do that my note will be updated at Zotero with related items in brackets. Is this possible to achieve in any way or does it jeopardize the integrity of my Zotero database?

Also I am concerned about the sync feature because I am using Zotero as vault for all my notes, readings and research. If this feature puts a risk on this database I am willing to let go Obsidian for good.

Edit: I also removed the replacement of spaces with hyphens from my note title template. Since I am not tech-savy, I wonder if it would affect any behaviour on Zotero side.

**Comment by windingwind** [comment]
> I want to make use of "related" field of Zotero to make the related items' title appear at the top of the note in brackets[[ ]]. This way, I can make Wikilinks via Zotero but when it syncs back it won't assume these links as related items

We could support updating Zotero note data after syncing back, but that's complicated since your `related` in Ob might not have a corresponding item in Z and we should define some extra rules for this, I am not sure we should take the risk of adding a lot of complexity to the project for this _tiny_ (that is, only used by a small group of users) feature? We don't support this currently and can have further discussion.

> If this feature puts a risk on this database I am willing to let go Obsidian for good.

AFAIK, no one has encountered this, there do exist some known issues about style problems between conversion of MD and Z-note, as you can see in the issues, which does not mess up your database but probably affects your current note/MD if you happen to have the case. That's not a serious problem and I will get them fixed one by one.

> I also removed the replacement of spaces with hyphens from my note title template. Since I am not tech-savy, I wonder if it would affect any behaviour on Zotero side.

If you mean the `ExportMDFilename` template, then it's OK and safe.

**Reply by ulahcherubim** [reply]
Thank you for the quick and extensive answer. Since two-way sync of related items is out of the window, is there a possibility for only getting related items in brackets at the top of the note? In this way I can make use of Zotero's related field and Obsidian's backlinks at the same time. I know this plugin offers bi-directional linking but I want to keep my workflow plain and ready for the long term without dependencies. Thanks again.

**Reply by ulahcherubim** [reply]
I have tried using double brackets on both Obsidian and Zotero sides and they are rendered with backslahes. /[/[
I am not sure about what is the reason for this behavior.

**Reply by windingwind** [reply]
I believe this Ob accent of Markdown is supported in a certain version above 1.0.4. What version are you on?

**Reply by ulahcherubim** [reply]
I am on version 1.1.4-49 in Zotero 7 Beta.

**Reply by windingwind** [reply]
See https://github.com/windingwind/zotero-better-notes/issues/744#issuecomment-1777106070

**Comment by ulahcherubim** [comment]
@windingwind Thank you for the reply and apologies for the late reply on my side. I know you do a lot of valuable work on the Zotero 7 and want to express my gratitude for that explicitly. 

Since updating related field from md file is a bit of a hussle for you to implement and keep up, is there a possibility to make it only work on Zotero to md export? Since I have decided to use solely Zotero for note taking and as a Zettelkasten, I want to make sure I can export all the note data I have inside the Zotero. If there was an option to edit metadata or note field by parsing the related items' title in double brackets, this would allow me to export and back up my notes easily. Heck, one can even export and print their noted with all tags and related items and use it as a physical Zettelkasten! (I am keeping a manual index and unique ID per note to keep that possibility as an option later)

I know I can make report from my collection to have the same result in a different way, so this not being implemented is not that big of a problem. Thanks again!
